pkg/apis/v1alpha1/common.go

```diff
@@ -33,6 +33,27 @@ type Param struct {
 	Value apiextensionsv1.JSON `json:"value"`
 }
 
+type DelegatableParam struct {
+	Name         string                `json:"name"`
+	Value        *apiextensionsv1.JSON `json:"value,omitempty"`
+	DefaultValue *apiextensionsv1.JSON `json:"default,omitempty"`
+}
+
+func (p *DelegatableParam) validateDelegatableParams() error {
+	if p.bothValuesSet() || p.neitherValueSet() {
+		return fmt.Errorf("invalid param: '%s', must set exactly one of value and default", p.Name)
```

Comment:
consider using brackets for param

Reply:
We'll do a robust search and replace for `'%?'` throughout the code in a later PR.